.clang-format

```diff
@@ -64,6 +64,8 @@ ForEachMacros:
   - foreach
   - Q_FOREACH
   - BOOST_FOREACH
+TypenameMacros: ['S2N_RESULT', 'S2N_CLEANUP_RESULT']
+AttributeMacros: ['S2N_RESULT_MUST_USE', 'S2N_API', 'S2N_PRIVATE_API']
```

Comment:
I would not include the changes to the api folder in this PR. The reformatting changes that come from adding S2N_API here are pretty significant, so it's better to keep that separate since this PR is about the testlib folder.

Reply:
The changes in the API folder are a result of a change in the clang-format file. I can split those changes into a new api PR. But the tls/extensions PR will be blocked on the new api PR. The block will happen because the api clang format checker will fail on the api folder without these changes.

Reply:
What exactly is the change in the clang-format file that caused this?

Reply:
AttributeMacros: 'S2N_API'. 

Clang-format is a parser only formater. I.e. it doesn't, for instance, run CPP. Therefore we need to tell it what various macro invocations mean. 

With Attribute Macros 'S2N_API' I'm saying to treat 'S2N_API' as an `__attribute__`. Clang's opinion is that these should be on the same line as the declaration. 

I've split this into it's own PR: https://github.com/aws/s2n-tls/pull/3663